src/sagemaker_xgboost_container/training.py

```diff
@@ -54,7 +55,16 @@ def run_algorithm_mode():
         checkpoint_config = {}
 
     train_path = os.environ[sm_env_constants.SM_CHANNEL_TRAIN]
-    val_path = os.environ.get(sm_env_constants.SM_CHANNEL_VALIDATION)
+    val_path = os.environ.get(sm_env_constants.SM_CHANNEL_VALIDATION,
+                              sm_env_constants.SM_CHANNEL_VALIDATION_NOT_SET)
+
+    if val_path != sm_env_constants.SM_CHANNEL_VALIDATION_NOT_SET:
```

Comment:
What's the advantage of using `SM_CHANNEL_VALIDATION_NOT_SET` over just checking for `None`? If `SM_CHANNEL_VALIDATION` does not exist in the environment, then `val_path` will be set to `None` in which case the check on 61 can just be `if val_path is not None:`.